### ISSUE TITLE
Hardcoded application names in the boosters

### ISSUE THREAD

**Issue by VeerMuchandi** [fabric8-launcher/launcher-planning#160]
All openshift objects (application name, service, etc) are all hardcoded in the templates. See this one https://github.com/wildfly-swarm-openshiftio-boosters/wfswarm-rest-http/blob/master/.openshiftio/application.yaml. When you use this booster it always creates the application with the name "wfswarm-rest-http". If you want to deploy multiple microservices they all get the same conflicting names on openshift!! This is a major issue using Launcher to deploy to openshift or minishift.

**Comment by gastaldi** [comment]
Add a parameter named `PROJECT` to the template and Launcher will set that for you

**Comment by VeerMuchandi** [comment]
How is that going to help? I want to set "ApplicationName" not "ProjectName".  Launcher is not even asking for application name. Is there a way to set ARTIFACT_ID that the launcher is capturing? I can use that as the application name instead.

**Comment by gastaldi** [comment]
PROJECT contains the Openshift project name that Launcher creates when you
specify it in the Project Info step.


I'll have a look if we can add more variables. Btw here is the part where
the template variables are applied:

https://github.com/fabric8-launcher/launcher-backend/blob/master/services/openshift-service-impl/src/main/java/io/fabric8/launcher/service/openshift/impl/Fabric8OpenShiftServiceImpl.java#L217

>

**Comment by gastaldi** [comment]
One solution would be to have a ProjectilePreparer implementation (like in  https://github.com/fabric8-launcher/launcher-backend/blob/master/core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/ChangeMavenMetadataPreparer.java) to replace the hardcoded name in the application with the Projectile artifactId.

You would end up with a hardcoded name in the booster OS template anyway, but at least that would contain the chosen artifactId instead.

**Comment by gastaldi** [comment]
Just to be clear we are talking about the same thing, what is "ApplicationName" in your context? I assume it is the project name when you type here or is it another thing that I am missing?

![screenshot from 2018-05-03 23-40-26](https://user-images.githubusercontent.com/54133/39611139-b0b6bd6c-4f2b-11e8-8f25-fd798ee35d32.png)

**Comment by gastaldi** [comment]
Also, if you deploy a REST HTTP booster  a Health Check booster in the same OpenShift project you would not get any name conflict afaik

**Comment by gastaldi** [comment]
Ah I think I understand now, maybe a solution in this case would be to replace the hardcoded `wfswarm-rest-http` with the Github repository name used to launch the Booster.

**Comment by VeerMuchandi** [comment]
if you are building multiple microservices that stay together in a project and more than one of them are using the same template, there is an issue. 

IMO, it is not a future consideration but an immediate need. When I create a microservice I want to name it what I want, not a name forced by the template. Without that launcher deploying to openshift directly is useless.

In the interim GitHub Repository name should be ok for the project name. Is it a quick change to use GitHub Repository name as a parameter here? 
https://github.com/fabric8-launcher/launcher-backend/blob/master/services/openshift-service-impl/src/main/java/io/fabric8/launcher/service/openshift/impl/Fabric8OpenShiftServiceImpl.java#L217

**Comment by gastaldi** [comment]
@VeerMuchandi I can change in the launcher-backend, but the boosters will have to be forked and changed to support the newly-created `SOURCE_REPOSITORY_NAME` parameter in https://github.com/fabric8-launcher/launcher-backend/issues/385. 

Given the time constraint, you'll need to fork those boosters and the booster-catalog repository and point your launcher installation to your forked booster-catalog repository if you want that feature before summit.

**Comment by gastaldi** [comment]
@VeerMuchandi I merged https://github.com/fabric8-launcher/launcher-backend/issues/385, can you give it a try by using it in a forked version of the `wf-swarm-rest-http` booster for example? 

I can give you further instructions to change your forked booster-catalog if you need.

**Comment by gastaldi** [comment]
@VeerMuchandi yes, as long as you keep the `fabric8/launcher-backend:latest` image tag in the template

**Comment by VeerMuchandi** [comment]
@gastaldi, I will test it.  A question though. What will be the value of SOURCE_REPOSITORY_NAME?  Is that set based on ArtifactId? 
Current it is being set as wfswarm-rest-http-redhat.

**Comment by gastaldi** [comment]
That is the GitHub repository name taken from the SOURCE_REPOSITORY_URL
used (the repository URL where the deployed booster lives)

Em Sex, 4 de mai de 2018 16:34, VeerMuchandi <notifications@github.com>
escreveu:

**Comment by VeerMuchandi** [comment]
Should it not use the GitHub Repository Name? If it uses a URL for the
application name will it not fail?

**Comment by VeerMuchandi** [comment]
@gastaldi: Should we rename this parameter SOURCE_REPOSITORY_NAME to APPLICATION_NAME instead? It can still be populated by the SOURCE_REPOSITORY_URL for now. The templates can expect a parameter with the name APPLICATION_NAME instead of SOURCE_REPOSITORY_NAME. This way when we change the launcher to take application name as the parameter, we don't have to come back and change all the templates.

**Comment by petenorth** [comment]
@gastaldi I just noticed the 'Try out the Beta version' link, this new wizard version does take an application name as input. Unfortunately I wasn't able to actually click the 'Set Up Application' button as the wizard seemed to think that both the mission and runtime sections were incomplete. This meant I wasn't able to confirm what the backend behaviour of the new wizard is.

One comment I would make is that the artifact id should default to the application name rather than booster but then you are faced with the question of what should be used as the default for the group id!

**Comment by gastaldi** [comment]
>I just noticed the 'Try out the Beta version' link, this new wizard version does take an application name as input. 

Yes, this is the OpenShift project name, we probably need to revisit this nomenclature with the UXD team

> Unfortunately I wasn't able to actually click the 'Set Up Application' button as the wizard seemed to think that both the mission and runtime sections were incomplete. This meant I wasn't able to confirm what the backend behaviour of the new wizard is.

You need to select the Mission and Runtime from a before proceeding (see screenshot below)

![screenshot from 2018-05-18 12-00-10](https://user-images.githubusercontent.com/54133/40242097-0c1a4e86-5a93-11e8-8959-c20e84b71fd4.png)

>One comment I would make is that the artifact id should default to the application name rather than booster

That's a great idea, I was talking to @quintesse yesterday and we thought we could save the users from asking this information and already suggest a name using a pattern like `${mission}-${runtime}` for the project and repository name. We need to validate with the UXD team first.

> what should be used as the default for the group id!

The default value for groupId is `io.openshift.booster`, but you can change that in the summary step.

**Comment by gastaldi** [comment]
> Should we rename this parameter SOURCE_REPOSITORY_NAME to APPLICATION_NAME instead? 

Ultimately I think we could use the application name (project name) that was specified when the wizard starts instead (right now the application name in the wizard means the name of the OpenShift project used when launching a booster).

**Comment by VeerMuchandi** [comment]
I think the project name and the application name are two different things
on openshift. You may have more than one application within a project.

**Comment by gastaldi** [comment]
@VeerMuchandi @petenorth sorry, I think I wasn't clear in my previous comment. 

I mean that in the UI the Application Name asked when you enter the wizard would be used as the Application name and we'd have another input `OpenShift Project Name` in the UI referring to the OpenShift project name to be used.

**Comment by VeerMuchandi** [comment]
+1 Peter, your example is pretty clear and that is exactly how I am seeing
it.
